tests/unit/datasets/test_synthetic.py

```diff
@@ -50,7 +50,8 @@ def test_tf_tensors_generation_cpu():
     assert tensors["user_id"].shape == (100, 1)
     assert tensors["user_age"].dtype == tf.float64
     for name, val in filter_dict_by_schema(tensors, schema.select_by_tag(Tags.LIST)).items():
-        assert len(val) == 2
+        assert isinstance(val, tf.SparseTensor)
```

Comment:
why is the dataloader now returning sparse tensors here. Previously every list feature, ragged or not was being returned by the loader as a tuple of (values, row_lengths).

Reply:
Thanks for catching this. It turns out this is due to the difference in default behaviors between the old merlin.models.loader and dataloader. The new dataloader uses the dataset schema and will create sparse tensors out of all list columns by default. In the old merlin.models.loader, the `sparse_*` parameters were not used by default (and it didn't use the dataset schema) and these were never used so we almost never expected sparse tensors. I think it's safe to keep the old behavior for now. I changed sparse tensors back to optional in [02a54d8](https://github.com/NVIDIA-Merlin/models/pull/845/commits/02a54d8a45ab95a4056b43157f53a44114763672).

Reply:
I've checked training a model with sparse tensors from the loader and results in the same issue described in #889 (the input signature loses it's names) if this did happen.